### PR TITLE
Adding reno to the repository

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

From now on Reno will be used to handle our changelog and we will lint
every PR to make.

### CODE REVIEW

File: docs/dev/contributing.md [hunk 1]
```diff
@@ -55,6 +56,23 @@ Please rebase your changes on `master` and squash your commits whenever possible
 it keeps history cleaner and it's easier to revert things. It also makes developers
 happier!
 
+### Reno
+
+We use `Reno` to create our CHANGELOG. Reno is a pretty simple
+[tool](https://docs.openstack.org/reno/latest/user/usage.html).
+
+Ulta quick `Reno` HOWTO:
```
Comment (olivielpeau) [inline docs/dev/contributing.md]: typo: `Ultra`

File: releasenotes/notes/add-reno-linting-to-test-23f722e6856fb1d8.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    We not handle release notes with Reno and lint every notes in the tests.
```
Comment (olivielpeau) [inline releasenotes/notes/add-reno-linting-to-test-23f722e6856fb1d8.yaml]: Can you "fix" this sentence? 😄

File: docs/dev/contributing.md [hunk 3]
```diff
@@ -55,6 +56,23 @@ Please rebase your changes on `master` and squash your commits whenever possible
...
+
+```bash
+$> reno new <topic-of-my-pr> --edit
+[...]
+# Remove unused section and fill the relevant ones
+# Reno will create a new file in releasenotes/notes
+[...]
+```
+
+Then just add and commit the new releasenote with your PR.
```
Comment (olivielpeau) [inline docs/dev/contributing.md]: can you mention that it's a `releasenote` _file_? :)

File: docs/dev/contributing.md [hunk 4]
```diff
@@ -55,6 +56,23 @@ Please rebase your changes on `master` and squash your commits whenever possible
 it keeps history cleaner and it's easier to revert things. It also makes developers
 happier!
 
+### Reno
```
Comment (olivielpeau) [inline docs/dev/contributing.md]: Can you add a very quick note on how to install reno in a dev env? (I guess mentioning `pip install reno` is enough)

**Review by olivielpeau — DISMISSED** [review]
A few nits on the docs, other than that LGTM!

I guess circleCI will be fixed as soon as #1000 is merged?

File: docs/dev/contributing.md (hunk 4 above)
Comment (hush-hush) [reply docs/dev/contributing.md]: good call !

**Comment by hush-hush** [comment]
@olivielpeau: yes I was waiting on #1000. It's working now !

**Review by olivielpeau — APPROVED** [review]
💯

**Comment by olivielpeau** [comment]
Approved!

Open question: what do you think would be a good way to remind contributors to write release notes for every PR (preferably before the CI fails)? A commit hook maybe?

**Comment by hush-hush** [comment]
Commit hooks never really works since they need to be installed on the client side. We should look into github API to see if a file was created under `releasenotes/note/` in the whole PR and not just a commit.

**Pull request merged**